File: ios/BluetoothGaming/Sources/BluetoothGaming/BluetoothGamingListener.swift
import Foundation

public protocol BluetoothGamingListener: AnyObject {
    func onPeerConnected(address: String, playerId: Int, displayName: String)
    func onPeerDisconnected(address: String)
    func onMessageReceived(address: String, messageType: UInt8, payload: Data)
    func onError(_ message: String)
}

public extension BluetoothGamingListener {
    func onPeerConnected(address: String, playerId: Int, displayName: String) {}
    func onPeerDisconnected(address: String) {}
    func onMessageReceived(address: String, messageType: UInt8, payload: Data) {}
    func onError(_ message: String) {}
}

File: android/bluetooth-gaming/src/main/kotlin/com/locxx/bluetoothgaming/BluetoothGamingListener.kt
package com.locxx.bluetoothgaming

interface BluetoothGamingListener {
    /** Every BLE advertisement while scanning (for manual host selection when UUID filtering fails). */
    fun onBleScanCandidate(address: String, name: String?, rssi: Int) {}

    fun onPeerConnected(address: String, playerId: Int, displayName: String) {}
    fun onPeerDisconnected(address: String) {}
    fun onMessageReceived(address: String, messageType: Byte, payload: ByteArray) {}
    fun onError(message: String) {}
}
